src/main/java/org/springframework/data/redis/connection/ReactiveStreamCommands.java

```diff
@@ -394,11 +396,40 @@ default Mono<RecordId> xAdd(ByteBufferRecord record) {
 		return xAdd(Mono.just(AddStreamRecord.of(record))).next().map(CommandResponse::getOutput);
 	}
 
+	/**
+	 * Add stream record with given {@literal body} to {@literal key} with specified options.
+	 *
+	 * @param record must not be {@literal null}.
+	 * @param xAddOptions additional parameters for the {@literal XADD} call, must not be {@literal null}.
+	 * @return {@link Mono} the {@link RecordId id}.
+	 * @see <a href="https://redis.io/commands/xadd">Redis Documentation: XADD</a>
+	 * @since 3.3
+	 */
+	default Mono<RecordId> xAdd(ByteBufferRecord record, XAddOptions xAddOptions) {
+
+		Assert.notNull(record, "Record must not be null");
+		Assert.notNull(xAddOptions, "XAddOptions must not be null");
+
+		AddStreamRecord addStreamRecord = AddStreamRecord.of(record)
+			.approximateTrimming(xAddOptions.isApproximateTrimming())
+			.makeNoStream(xAddOptions.isNoMkStream());
+
+		if (xAddOptions.hasMaxlen()) {
+			addStreamRecord = addStreamRecord.maxlen(xAddOptions.getMaxlen());
+		}
+
+		if (xAddOptions.hasMinId()) {
+			addStreamRecord = addStreamRecord.minId(xAddOptions.getMinId());
+		}
+
+		return xAdd(Mono.just(addStreamRecord)).next().map(CommandResponse::getOutput);
+	}
```

Comment:
the reason why I wrote somewhat verbose mehtod in here is

unlike `JedisStreamCommands` or `LettuceStreamCommands` 
which both have a method accepting (MapRecord<byte[], byte[], byte[]> record, XAddOptions options)

xAdd(Publisher<AddStreamRecord> commands) in `LettuceReactiveStreamCommands`
expects that Publisher<AddStreamRecord> already has variables of `XAddOptions`. 
ex) maxlen, isApproxiateTrimming...

so I placed the decapsulation of XAddOptions here.

I thought of changing AddStreamRecord to have variable `priavte final XAddOptions options`
so that `LettuceReactiveStreamCommands` can delegate the decapsulation to other class
but I thought this exceeds the scope of this issue.